neonvm/hack/kernel/Dockerfile.kernel-builder

```diff
@@ -1,11 +1,14 @@
-FROM ubuntu:23.10 AS build-deps
-
+# Force initial check that KERNEL_VERSION is set appropriately
+FROM ubuntu:23.10 AS check-arg
 ARG KERNEL_VERSION
+WORKDIR /build
 
 RUN set -e \
-    && echo "Build linux  kernel ${KERNEL_VERSION}" \
-    && test -n "${KERNEL_VERSION}"
+    && echo "Build linux kernel ${KERNEL_VERSION}" \
+    && test -n "${KERNEL_VERSION}" \
+    && echo "force this as a requirement for build-deps" > /build/arg-check-succeeded
```

Comment:
Why not just put this part after the apt-get?

Reply:
We could. I implemented it this way because I liked that previously, if there was an issue with `KERNEL_VERSION`, it would show up right away -- instead of needing to wait for `apt-get install` to run first (which can take a while).